src/main/java/com/firebolt/jdbc/metadata/MetadataUtil.java

```diff
@@ -102,4 +102,14 @@ public static String getDatabaseVersionQuery(String engine) {
 				.conditions(Collections.singletonList(String.format("engine_name iLIKE '%s%%'", engine))).build()
 				.toSql();
 	}
+
+	// lower() is used in SQL statement because the comparison in Firebolt is case-sensitive but we retrieve engine name from host where lowercase is used.
+	// But default engine is created with suffix Ingest (upper I). This is the reason to call loser() function.
+	public static String isConnectionReadOnly() {
+		return "select count(*) = 0 from information_schema.engines where status = 'Running' and engine_type = 'General Purpose' and lower(engine_name) = lower(?)";
+	}
+
+	public static String isDatabaseReadOnly() {
+		return "select count(*) = 0 from information_schema.engines where status = 'Running' and engine_type = 'General Purpose' and lower(attached_to) = lower(?)";
+	}
```

Comment:
This is fine for Firebolt 1.0, but just a heads-up - FB 2.0 will have RBAC so cases when a user doesn't have access to GP engine or user is set to reader despite working on a GP engine will be possible.

Reply:
Well... Do you have better solution? I will appreciate any help.

Reply:
Don't know. You'll need to check with the relevant team if there's a way to verify access. #sql-frontent will probably be a good place to start. This is the only driver that exposes `is_ReadOnly` so we haven't had the same issue elsewhere.
Now that V1 and V2 are merged `is_ReadOnly` will potentially return false info with RBAC so I'm not sure if we should implement it in the current state.